File: 1-two-sum/1-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map <int, int> Mp;
        for(int i = 0; i < nums.size(); i++) {
            if(Mp.find(target - nums[i]) != Mp.end()) {  
                return {Mp[target - nums[i]], i};
            }
            Mp[nums[i]] = i;   
        }
        return {};
    }
};

File: 1-two-sum/1-two-sum.swift
class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        var hashSet = [Int: Int]()
        for(index, item) in nums.enumerated() {
            let x = target - item
            if let val = hashSet[x] {
                return [val, index]
            }
            hashSet[item] = index
        }
        return []
    }
}